findSurfaceResiduesListCharged.py

```diff
@@ -0,0 +1,195 @@
+
+USAGE
+
+    findSurfaceResiduesListCharged [pH, [selection ,[cutoff ,[ doShow , [ quiet]]]]]]
+
+ARGUMENTS
+
+    pH = The pH value to estimate a surface charge at
+
+    selection = string: object or selection in which to find exposed
+    residues {default: all}
+
+    cutoff = float: cutoff of what is exposed or not {default: 2.5 Ang**2}
+
+RETURNS
+
+    (list: (chain, resv ) )
+        A Python list of residue numbers corresponding
+        to those residues w/more exposure than the cutoff.
+
+    A printout of all charged amino acids and the estimated surface charge of a protein.
+
+    """
+    cutoff, doShow, quiet = float(cutoff), int(doShow), int(quiet)
+
+    selName = findSurfaceAtoms(selection, cutoff, quiet)
+
+    exposed = set()
+    cmd.iterate(selName, "exposed.add((chain,resv))", space=locals())
+
+    selNameRes = cmd.get_unused_name("exposed_res_")
+    cmd.select(selNameRes, "byres " + selName)
+
+    if not quiet:
+        print("Exposed residues are selected in: " + selNameRes)
+
+    if doShow:
+        cmd.show_as("spheres", "(" + selection + ") and polymer")
+        cmd.color("white", selection)
+        cmd.color("yellow", selNameRes)
+        cmd.color("red", selName)
+    exposed=sorted(exposed) #list of exposed residues
+    seq=cmd.get_fastastr('all')
+    newseq = seq.replace('\n', ' ').replace('\r', '').replace(' ','').replace("?","")
+    nDomains=newseq.count(">")
+
+    def splitSeq(seq,d): #splits residues into their domains
+        domain=-1
+        count=0
+        splits=[""]*d
+        for r in seq:
+            if r==">":
+                domain+=1
+                count=7
+            if count>0:
+                count-=1
+                continue
+            if r=="\n":
+                continue
+            splits[domain]+=r
+        return splits
+
+    splits=splitSeq(newseq,nDomains) #list of all residues separated by domain
+
+    #gets all charged amino acids on the surface
+    reslist=[[] for x in range(nDomains)]
+    exposedAms=""
+    K=0
+    R=0
+    D=0
+    H=0
+    E=0
+
+    for r in exposed:
+        tempDomain=ord(r[0])-65
+        if r[1]-1 >= len(splits[tempDomain]):
+            continue
+        amino=splits[tempDomain][r[1]-1]
+        if amino not in "KRDHE":
+            continue
+        elif amino=='K':
+            K+=1
+        elif amino=='R':
+            R+=1
+        elif amino=='D':
+            D+=1
+        elif amino=="H":
+            H+=1
+        elif amino=='E':
+            E+=1
+        exposedAms+=amino
+        amPos=amino+str(r[1])
+        reslist[tempDomain].append(amPos)
+
+    n=1
+    for dom in reslist:
+        print("Domain " + str(n)+" charged residues:")
+        print(dom)
+        n+=1
+
+    print("Charged residue list")
+    print(exposedAms)
+    print("Number of exposed charged residues")
+    print(len(exposedAms))
+
+    if pH > 10.54:
+        kCharge=10 ** -(pH-10.54)
+    else:
+        kCharge=1-(10** (pH-10.54))
```

Comment:
I think this math is wrong. Check for example [Biopython](https://github.com/biopython/biopython/blob/0c7cd95d22ac08525d17adb838fd2802200d1f11/Bio/SeqUtils/IsoelectricPoint.py#L114-L133) or [propka](https://github.com/jensengroup/propka/blob/68007f065c5400566598509875344ed1c0f593c9/propka/group.py#L586-L602) which use the Henderson Hasselbalch equation.